pkg/logging/writer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+package logging
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/rs/zerolog"
+)
+
+// Writer implements LogSink interface using zerolog.Logger
+
+type Writer struct {
+	Log       *zerolog.Logger
+	Verbosity zerolog.Level
+	Caller    string
+}
+
+// ignoring CallDepth
+func (w *Writer) Init(ri logr.RuntimeInfo) {
+}
+
+func (w *Writer) Enabled(lvl int) bool {
+	return lvl >= int(w.Verbosity)
+}
+
+func (w *Writer) WithValues(keysAndValues ...interface{}) logr.LogSink {
+	values, err := handleFields(keysAndValues)
+	if err != nil {
+		w.Log.Error().Msg(err.Error())
+		return w
+	}
+	l := w.Log.With()
+	for key, val := range values {
+		l = l.Str(key, val)
+	}
+	newLog := l.Logger()
+	return &Writer{Log: &newLog, Verbosity: w.Verbosity, Caller: w.Caller}
+}
+
+func (w *Writer) Info(lvl int, msg string, keysAndVals ...interface{}) {
+	if !w.Enabled(lvl) {
+		return
+	}
+	values, err := handleFields(keysAndVals)
+	if err != nil {
+		w.Log.Error().Msg(err.Error())
+	}
+	l := w.Log.Info()
+	if w.Caller != "" {
+		l = l.Str(SETUP, w.Caller)
+	}
+	for key, val := range values {
+		l = l.Str(key, val)
+	}
+	l.Msg(msg)
```

Comment:
you probably should replace the line to `l.CallerSkipFrame(3).Msg(msg)`. Otherwise, the caller's line will be this line. 
The problem that internally sometimes we need to skip 2 frames, and sometimes 3. 
I suggest skipping 3 frames, at least it helps to find the line where the method was called from.

Reply:
Added skip level of 2 plus support of caller depth inside the writer implementation. Usually the internal implementation adds additional depth of 1. The results match the klog/zap caller info.